src/utils/errors.ts

```diff
@@ -0,0 +1,32 @@
+import { Transactions } from ".prisma/client";
+import { getChainByChainIdAsync } from "@thirdweb-dev/chains";
+import { ethers } from "ethers";
+
+interface EthersError {
+  reason: string;
+  code: string;
+  error: any;
+  method: string;
+  transaction: any;
+}
+
+export const parseTxError = async (
+  tx: Transactions,
+  err: any,
+): Promise<string | null> => {
+  if (!err) {
+    return null;
+  }
+
+  if ((err as EthersError)?.code === ethers.errors.INSUFFICIENT_FUNDS) {
```

Comment:
Basically maps the insufficient funds error code to a more human-readable error message.

Reply:
I logged `err` :

```
{
  code: 3,
  message: 'insufficient funds for gas * price + value: balance 1636106304978556, queued cost 1570206001522624, tx cost 1570206001522624, overshot 1504305698066692'
}
```

and looks like `err.code` is `3` when the `insufficient ...` message comes from RPC and `ethers.errors.INSUFFICIENT_FUNDS` is `INSUFFICIENT_FUNDS`, so they check fails. though the RPC message is elaborate but the balance value seems off.

also, when wallet-balance is low, gas-estimation fails too:

```Error: cannot estimate gas; transaction may fail or may require manual gas limit [ See: https://links.ethers.org/v5-errors-UNPREDICTABLE_GAS_LIMIT ] (error={"reason":"cannot estimate gas; transaction may fail or may require manual gas limit","code":"UNPREDICTABLE_GAS_LIMIT","error":{"code":3},"method":"estimateGas","transaction":{"from":"0x3EcDBF3B911d0e9052b64850693888b008e18373","maxPriorityFeePerGas":{"type":"BigNumber","hex":"0x07aef40a00"},"maxFeePerGas":{"type":"BigNumber","hex":"0x07aef40a20"},"to":"0x365b83D67D5539C6583b9c0266A548926Bf216F4","value":{"type":"BigNumber","hex":"0x00"},"data":"0xa9059cbb0000000000000000000000003ecdbf3b911d0e9052b64850693888b008e1837300000000000000000000000000000000000000000000000000766c7d74835555","type":2,"accessList":null}}, tx={"to":{},"from":"0x3ecdbf3b911d0e9052b64850693888b008e18373","data":"0xa9059cbb0000000000000000000000003ecdbf3b911d0e9052b64850693888b008e1837300000000000000000000000000000000000000000000000000766c7d74835555","value":{"type":"BigNumber","hex":"0x00"},"nonce":{"type":"BigNumber","hex":"0x1836"},"maxFeePerGas":{"type":"BigNumber","hex":"0x07aef40a20"},"maxPriorityFeePerGas":{"type":"BigNumber","hex":"0x07aef40a00"},"type":2,"gasLimit":{},"chainId":{}}, code=UNPREDICTABLE_GAS_LIMIT, version=abstract-signer/5.7.0)```